### PR TITLE
validator: coerction float from int for type float

### PR DESCRIPTION
**Pull request by lysyjan**

### CODE REVIEW

File: src/Validator.php [hunk 1]
```diff
@@ -50,6 +50,9 @@ class Validator
 	/** @var bool */
 	public $coerceStringToDateTimeImmutable = false;
 
+	/** @var bool */
+	public $coerceIntToFloat = false;
```
Comment (hrach) [inline src/Validator.php]: please move this to coerceStringToFloat

File: src/Validator.php [hunk 2]
```diff
@@ -142,6 +145,10 @@ public function validate($data): Result
 							$node = $filteredValue;
 							$isValid = $this->validateNumber($node, $schema, $path, $errors);
 							break;
+						} elseif ($this->coerceIntToFloat && is_int($node) && ($filteredValue = filter_var($node, FILTER_VALIDATE_FLOAT)) !== false) {
```
Comment (hrach) [inline src/Validator.php]: the filter_var call seem to be completely unnecessary.

File: tests/cases/validate.float.phpt [hunk 3]
```diff
@@ -102,6 +103,10 @@ Assert::true($result->isValid());
 Assert::same(49.210420445650314, $result->getData());
 Assert::same('49.21042044565', (string) $result->getData()); // precission loss
 
+$result = $validator->validate('52');
```
Comment (hrach) [inline tests/cases/validate.float.phpt]: this is a wrong  test, it tests coercing a string, not an int.

**Review by hrach — CHANGES_REQUESTED** [review]

**Comment by lysyjan** [comment]
updated

**Comment by hrach** [comment]
👍

**Pull request merged**